docs: multi modalities (#1317)
docarray/docarray: docs/data_types/audio/audio.md

# 🔊 Audio

DocArray supports many different modalities including `Audio`.
This section will show you how to load and handle audio data using DocArray.

> Moreover, we will introduce DocArray's audio specific types, to represent your audio data ranging from [`AudioUrl`][docarray.typing.url.AudioUrl] to [`AudioBytes`][docarray.typing.bytes.AudioBytes] and [`AudioNdArray`][docarray.typing.tensor.audio.audio_ndarray.AudioNdArray].

Reviewer: ```suggestion
Moreover, you will learn about DocArray's audio specific types, to represent your audio data ranging from [`AudioUrl`][docarray.typing.url.AudioUrl] to [`AudioBytes`][docarray.typing.bytes.AudioBytes] and [`AudioNdArray`][docarray.typing.tensor.audio.audio_ndarray.AudioNdArray].
```

!!! note
    This requires a `pydub` dependency. You can install all necessary dependencies via:
    ```cmd 
    pip install "docarray[audio]"
    ```

## Load audio file

First, let's define a class, which extends [`BaseDoc`][docarray.base_doc.doc.BaseDoc] and has an `url` attribute of type [`AudioUrl`][docarray.typing.url.AudioUrl], and an optional `tensor` attribute of type `AudioTensor`.

> Next, we can instantiate an object of that class with a local or remote url. 

Reviewer: ```suggestion
Next, you can instantiate an object of that class with a local or remote URL. 
```

```python
from docarray import BaseDoc
from docarray.typing import AudioUrl, AudioNdArray

class MyAudio(BaseDoc):
    url: AudioUrl
    tensor: AudioNdArray = None
    frame_rate: int = None

doc = MyAudio(
    url='https://github.com/docarray/docarray/blob/feat-rewrite-v2/tests/toydata/hello.mp3?raw=true'
)

> ```

Reviewer: Should we have a `note` or something that referrs to our pre-built AudioDoc?

- the corresponding frame rate describing the number of signals for a certain period of time

```python
doc.tensor, doc.frame_rate = doc.url.load()
doc.summary()
```
<details>
    <summary>Output</summary>
    ``` { .text .no-copy }
    📄 MyAudio : 2015696 ...
    ╭──────────────────────┬───────────────────────────────────────────────────────╮
    │ Attribute            │ Value                                                 │
    ├──────────────────────┼───────────────────────────────────────────────────────┤
    │ url: AudioUrl        │ https://github.com/docarray/docarray/blob/feat-rew    │
    │                      │ ... (length: 90)                                      │
    │ tensor: AudioNdArray │ AudioNdArray of shape (30833,), dtype: float64        │
    │ frame_rate: int      │ 44100                                                 │
    ╰──────────────────────┴───────────────────────────────────────────────────────╯
    ```
</details>

## AudioTensor

DocArray offers several AudioTensors to store your data to:

- [`AudioNdArray`][docarray.typing.tensor.audio.audio_ndarray.AudioNdArray]
- `AudioTorchTensor`

> - `AudioTensorFlowTensor`

Reviewer: Can we also link to the torch and tf variants? why does only the np one have the link?

)

doc.bytes_, _ = doc.url.load_bytes()  # type(doc.bytes_) = AudioBytes
doc.tensor, _ = doc.bytes_.load()  # type(doc.tensor) = AudioNdarray
```
 
Vice versa, you can also transform an [`AudioTensor`] to [`AudioBytes`][docarray.typing.bytes.AudioBytes]:

```python
from docarray.typing import AudioBytes

bytes_from_tensor = doc.tensor.to_bytes()

assert isinstance(bytes_from_tensor, AudioBytes)
```

## Save audio to file
You can save your AudioTensor to an audio file of any format as follows:
```python
tensor_reversed = doc.tensor[::-1]
tensor_reversed.save(
    file_path='olleh.mp3',
    format='mp3',
)
```
## Play audio in notebook

You can play your audio sound in a notebook from its url as well as its tensor, by calling `.display()` on either one.

Play from `url`:
```python
doc.url.display()
```

<table>
  <tr>
    <td><audio controls><source src="../hello.mp3" type="audio/mp3"></audio></td>
  </tr>
</table>

Play from `tensor`:
```python
tensor_reversed.display()
```
<table>
  <tr>
    <td><audio controls><source src="../olleh.mp3" type="audio/mp3"></audio></td>
  </tr>
</table>

> ## Getting started - Predefined `AudioDoc`

Reviewer: Mayb we can link to this section where you first create the custom audio doc
